onnxruntime/core/framework/utils.cc

```diff
@@ -448,7 +457,9 @@ static void FinalizeFeedFetchCopyInfo(FeedsFetchesManager& feeds_fetches_manager
     if (feed.IsTensor()) {
       feed_locations[i] = feed.Get<Tensor>().Location().device;
     } else if (feed.IsSparseTensor()) {
+#if !defined(DISABLE_SPARSE_TENSORS)
       feed_locations[i] = feed.Get<SparseTensor>().Location().device;
+#endif
     }
```

Comment:
Does IsSparseTensor throw if sparse tensor support is diabled so we don't need to do anything here to prevent silent failure? If not it's probably easier to do it there vs. at the calling sites.

Reply:
`IsSparseTensor` is specified as `noexcept`, so might not be able to do a throw there:(

Reply:
Can you remove the noexcept?